### PR TITLE
Implement Pokemon.typing

### PR DESCRIPTION
**Pull request by AustinXII**
also fix multiscale and abilities calc

### CODE REVIEW

**Comment by AustinXII** [comment]
Oh I guess it would have been better to just fix that last PR

File: js/ap_calc.js [hunk 1]
```diff
@@ -343,6 +343,7 @@ function Pokemon(pokeInfo) {
         var pokemon = pokedex[this.name];
         this.type1 = pokemon.t1;
         this.type2 = (pokemon.t2 && typeof pokemon.t2 !== "undefined") ? pokemon.t2 : "";
+        this.typing = this.type1 || this.type2;
```
Comment (Zarel) [inline js/ap_calc.js]: This is still not how programming works...

**Comment by MagiaSN** [comment]
According to my understanding, because both `type1` and `type2` are string (not `null` or `undefined`) and `type1` is never empty, `type1 || type2` would simply returns `type1`, so is seems this may not work

**Comment by Zarel** [comment]
^

Yes, that's what I meant when I said "This is still not how programming works..."

**Comment by Honko** [comment]
I think what you're aiming for is a hasTyping function, which sounds like a good idea to me.

**Comment by AustinXII** [comment]
@Honko yeah and I'll get to this eventually

**Pull request closed without merging**